pb/edge_ctrl_pb/edge_ctrl.proto

```diff
@@ -3,182 +3,240 @@ syntax = "proto3";
+  PostureQuery.Type type = 40;
+  string version = 50;
+  oneof parameters {
+    PostureQuery.None na = 60;
+    PostureQuery.Process process = 70;
 
-    }
+  }
 
-    enum Type {
-        OS = 0;
-        MAC = 10;
-        WINDOWS_DOMAIN = 20;
-        PROCESS = 30;
-    }
+  enum Type {
+    OS = 0;
+    MAC = 10;
+    WINDOWS_DOMAIN = 20;
+    PROCESS = 30;
+  }
 
-    enum HashAlgorithm {
-        SHA512 = 0;
-    }
+  enum HashAlgorithm {
+    SHA512 = 0;
+  }
 
-    message None {
+  message None {
 
-    }
+  }
 
-    message Process {
-        repeated string paths = 1;
-        HashAlgorithm hashAlgorithm = 10;
-    }
+  message Process {
+    repeated string paths = 1;
+    HashAlgorithm hashAlgorithm = 10;
+  }
 }
 
 
 message PostureResponse {
-    string id = 1;
-    oneof payload {
-        PostureResponse.Error error = 20;
-        PostureResponse.Proc process = 30;
-        PostureResponse.MacAddresses mac = 40;
-        PostureResponse.WindowsDomain domain = 50;
-        PostureResponse.OperatingSystem os = 60;
+  string id = 1;
+  oneof payload {
+    PostureResponse.Error error = 20;
+    PostureResponse.Proc process = 30;
+    PostureResponse.MacAddresses mac = 40;
+    PostureResponse.WindowsDomain domain = 50;
+    PostureResponse.OperatingSystem os = 60;
+  }
+
+
+  message Error {
+    enum Code {
+      UNKNOWN = 0;
+      UNSUPPORTED_QUERY_TYPE = 10;
+      UNSUPPORTED_HASH_ALGORITHM = 20;
     }
 
+    PostureResponse.Error.Code code = 1;
+    string message = 10;
+    map<string, string> data = 20;
+  }
 
-    message Error {
-        enum Code {
-            UNKNOWN = 0;
-            UNSUPPORTED_QUERY_TYPE = 10;
-            UNSUPPORTED_HASH_ALGORITHM = 20;
-        }
+  message Proc {
+    string binaryHash = 1;
+    string signerFingerprint = 10;
+  }
 
-        PostureResponse.Error.Code code = 1;
-        string message = 10;
-        map<string, string> data = 20;
-    }
+  message MacAddresses {
+    repeated string addresses = 1;
+  }
 
-    message Proc {
-        string binaryHash = 1;
-        string signerFingerprint = 10;
-    }
+  message WindowsDomain {
+    string name = 1;
+  }
 
-    message MacAddresses {
-        repeated string addresses = 1;
-    }
+  message OperatingSystem {
+    PostureResponse.OperatingSystem.Type type = 1;
+    string version = 10;
+    string build = 20;
 
-    message WindowsDomain {
-        string name = 1;
+    enum Type {
+      Android = 0;
+      iOS = 10;
+      Linux = 20;
+      macOS = 30;
+      Windows = 40;
+      WindowsServer = 50;
     }
+  }
+}
 
-    message OperatingSystem {
-        PostureResponse.OperatingSystem.Type type = 1;
-        string version = 10;
-        string build = 20;
-
-        enum Type {
-            Android = 0;
-            iOS = 10;
-            Linux = 20;
-            macOS = 30;
-            Windows = 40;
-            WindowsServer = 50;
-        }
-    }
+message CreateCircuitRequest {
+  string sessionToken = 1;
```

Comment:
This message should include `sessionId` to be used on the receiving end for logging.

Reply:
It's not available from the edge router to send.